superset/connectors/sqla/models.py

```diff
@@ -1790,23 +1806,448 @@ def before_update(
             raise Exception(get_dataset_exist_error_msg(target.full_name))
 
     @staticmethod
-    def update_table(
-        _mapper: Mapper, _connection: Connection, obj: Union[SqlMetric, TableColumn]
+    def update_table(  # pylint: disable=unused-argument
+        mapper: Mapper, connection: Connection, target: Union[SqlMetric, TableColumn]
     ) -> None:
         """
         Forces an update to the table's changed_on value when a metric or column on the
         table is updated. This busts the cache key for all charts that use the table.
 
-        :param _mapper: Unused.
-        :param _connection: Unused.
-        :param obj: The metric or column that was updated.
+        :param mapper: Unused.
```

Comment:
Would it make sense to pull this hook logic out into each "New" model's respective classes? This is a rather large block and could use a little breakup :).

Reply:
I left it in the old model because it will go away with the old model — this is only needed while both exist. But I agree it's hard to read, I'll break it down. There's also some duplicate logic, but since this is temporary code that's going away I thought it would be ok.